helpers/decrementVar.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+const utils = require('handlebars-utils');
+const max_keys = 50;
+
+const factory = globals => {
+    return function(key) {
+        if (!utils.isString(key)) {
+            throw new Error("decrementVar helper key must be a string");
+        }
+
+        // Setup storage
+        if (typeof globals.storage.variables === 'undefined') {
+            globals.storage.variables = {};
+        }
+
+        // Make sure the number of total keys is within the limit
+        if (Object.keys(globals.storage.variables).length >= max_keys) {
+            throw new Error(`Unique keys in variable storage may not exceed ${max_keys} in total`);
+        }
+
+        if (typeof globals.storage.variables[key] !== 'undefined'
+            && Number.isInteger(globals.storage.variables[key])) {
```

Comment:
You can probably simplify this since `Number.isInteger` will return false if value is undefined. So maybe

```
if (Number.isInteger(globals.storage.variables[key]) {
  // decrement
} else if (typeof globals.storage.variables[key] === 'undefined') {
  // create variable
} else {
  // throw error
}
```

I would also think about changing this so we throw an error if incrementing or decrementing a non-existent value. It's not clear to me that calling `decrementVar` on a nonexistent value should set it to zero.